tokio/src/runtime/scheduler/multi_thread/worker.rs

```diff
@@ -612,7 +616,33 @@ impl Core {
         if self.tick % worker.handle.shared.config.global_queue_interval == 0 {
             worker.inject().pop().or_else(|| self.next_local_task())
         } else {
-            self.next_local_task().or_else(|| worker.inject().pop())
+            let maybe_task = self.next_local_task();
+
+            if maybe_task.is_some() {
+                return maybe_task;
+            }
+
+            let cap = self.run_queue.capacity();
```

Comment:
The documentation of this method should probably call out that you can only _decrease_ the amount of available slots by accessing the `run_queue`, to help make it clear why `push_back` can't fail.

Reply:
Done